backup/Command1.java

```diff
@@ -0,0 +1,23 @@
+public class Command {
+    protected CommandType type;
+    protected int index;
```

Comment:
I really liked the way you include attributes of classes as protected. The concept of encapsulation and abstraction is well mastered.